### PR TITLE
CPR-381-change-person-identifier-table

### PR DESCRIPTION
**Pull request by JunaedAhmedMOJ**
changed name of person identifier table
refactored code referencing previous db name

### CODE REVIEW

**Review by markreesmoj — APPROVED** [review]
👍

File: src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/jpa/repository/specifications/queries/PersonSpecificationQueries.kt [hunk 1]
```diff
@@ -30,7 +30,7 @@ private fun findCandidates(person: Person): Specification<PersonEntity> {
 }
 
 fun findCandidatesWithUuid(person: Person): Specification<PersonEntity> {
-  return findCandidates(person).and(PersonSpecification.hasPersonIdentifier())
+  return findCandidates(person).and(PersonSpecification.haspersonKey())
```
Comment (H-Iwanejko) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/jpa/repository/specifications/queries/PersonSpecificationQueries.kt:33]: small one should be `hasPersonKey()`

Comment (JunaedAhmedMOJ) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/jpa/repository/specifications/queries/PersonSpecificationQueries.kt]: good spot :) PR updated

File: src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/PersonService.kt [hunk 2]
```diff
@@ -109,23 +109,23 @@ class PersonService(
 
   private fun isCreateEvent(event: String?) = listOf(PRISONER_CREATED, NEW_OFFENDER_CREATED).contains(event)
 
-  private fun createPersonIdentifier(person: Person): PersonIdentifierEntity {
-    val personIdentifier = PersonIdentifierEntity.new()
+  private fun createpersonKey(person: Person): PersonKeyEntity {
```
Comment (markreesmoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/PersonService.kt:112]:
```suggestion
  private fun createPersonKey(person: Person): PersonKeyEntity {
```

File: src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/PersonService.kt [hunk 3]
```diff
@@ -109,23 +109,23 @@ class PersonService(
...
       person,
-      mapOf(EventKeys.UUID to personIdentifier.personId.toString()),
+      mapOf(EventKeys.UUID to personKey.personId.toString()),
     )
-    return personIdentifierRepository.saveAndFlush(personIdentifier)
+    return personKeyRepository.saveAndFlush(personKey)
   }
 
-  private fun retrievePersonIdentifier(person: Person, personEntity: PersonEntity): PersonIdentifierEntity {
+  private fun retrievepersonKey(person: Person, personEntity: PersonEntity): PersonKeyEntity {
```
Comment (markreesmoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/PersonService.kt:122]:
```suggestion
  private fun retrievePersonKey(person: Person, personEntity: PersonEntity): PersonKeyEntity {
```

**Review by markreesmoj — COMMENTED** [review]
I am not totally in love with personKey as a name but it is better than personIdentifier. There are a couple of places where we need to get the casing right, otherwise this looks good

File: src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/PersonService.kt (hunk 3 above)
Comment (JunaedAhmedMOJ) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/PersonService.kt]: updated :)

**Review by markreesmoj — APPROVED** [review]
👍

**Pull request merged**